Conda Parser - Solve conda environments by an api call to /parse

tyrelsouza opened the pull request:
All python code is formatted with `black` so style comments [will be ignored.](https://www.mattlayman.com/blog/2018/python-code-black/)

Note - docker pull stuff doesn't work yet, Going to get that up in the docker regs on Monday.

tyrelsouza committed:
Conda Parser
fix gitignore
add more docstring
redo pip requirements.txt
environment
Readme update miniconda and link
docker-compose isort black bump python 3.7.4
default port 5000
Readme rearrange
leftover cleanup:
rename start.sh to gunicorn_start.sh and run.py to flask_start.py
License me captain
.travis.yml
python 3.7.4
redo Dockerfile, copy twice so cache invaliation doesnt happen as frequent
ignore all dicts, not just pip
only check strings
four spaces, oops my ruby is leaking
add better comment
explicitly only the files we need in image
docker comments

tyrelsouza commented on conda_parser/parse.py:
```diff
@@ -0,0 +1,100 @@
+        )  # Default Requirement. Appending because no requirement specified, will get sliced into return.
+        name, requirement = parts[:2]
+
+    return {"name": name, "requirement": requirement}
```
This method is where all the version parsing logic is. It's not as great as I want it, because I'm not spinning up new conda envs and throwing them away, so this varies by what people pass in.

tyrelsouza replied:
That will come with the next PR which I'm going to start as soon as this is Merged.

jsonperl commented on docker-compose.yml:
```diff
@@ -0,0 +1,9 @@
+    build: .
+    ports:
+    - "5000:5000"
+    volumes:
```
I think this is going to override the data you copied in the Dockerfile. What's the goal with the volume?

tyrelsouza replied:
Discussing quicker on Company slack, and will come back and post resolution for posterity.

tyrelsouza replied:
Copying `environment.yml` earlier so that the cache is not invalidated as frequently.

mikeyoung85 commented on conda_parser/parse.py:
```diff
@@ -0,0 +1,110 @@
+    dependencies = []
+
+    for dep in raw_dependencies:
+        if isinstance(dep, dict) and dep.get("pip"):
```
Are there other cases where a dependency would be a dictionary outside of pip that we would care about?

tyrelsouza replied:
Not at this moment. @havocp pointed me at this https://github.com/Anaconda-Platform/anaconda-project/blob/dd4d81fc266d3ba5d34bc7fb0f0a85c8eafbbf40/anaconda_project/env_spec.py#L438-L491

tyrelsouza committed:
conda api Solver

remove v1, just always present version, parse with conda api
update requirements
import sort and black
double timeout time
conda ver diff
requirements.txt
play with numbers
yolo git egg
don't pin this

tyrelsouza committed:
Docker tests

start tests command, pytest.ini for skipping src/
no isort, no editable egg, oops
make testable with docker
testing travis docker
no 3.7.4 on travis...
3.7 maybe?
change linux distro, and fix build script
try pytest without pip install..
not needed
test from docker only, oops

tiegz commented on .gitignore:
```diff
@@ -0,0 +1,67 @@
+
+
+
+
```
Extremely minor, but a bunch of extra newlines.

tyrelsouza replied:
weird. gitignore.io does what it wants I guess.

tiegz commented on README.md:
```diff
@@ -0,0 +1,89 @@
+Then, run:
+
+    $ docker pull librariesio/conda-parser
+    $ docker run -it -e PORT=5000 -p 5000:5000 librariesio/conda-parser
```
Looks like this image isn't pushed/public yet?

tyrelsouza replied:
(it is not, there's a note in the PR body about this)

tiegz reviewed (commented):
Nice!!

Q: I looked up some sample files on GH and found this one. It throws a 500 because of the `pip` sub-section. Is that valid syntax? https://raw.githubusercontent.com/SilvioGiancola/SoccerNet-code/552522b4122eac600a02ba6b28cd7c5b0c5d893e/src/feature_extraction/i3d-feat-extract/environment.yml

tyrelsouza committed:
fixing bad styling

tyrelsouza committed:
Skip parsing Pip dependencies for now.

tyrelsouza committed:
vim is removing trailign newlines, black removes them, I need to setup a pre-commit

tyrelsouza committed:
Merge pull request #3 from librariesio/tyrelsouza/bring-back-no-pip

Skip pip.
